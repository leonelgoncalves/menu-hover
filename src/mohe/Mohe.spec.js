import React from "react";
import { mount, render } from "enzyme";
import MoheComponent from "./mohe";

describe("MoheComponent:", () => {
  const setup = propOverrides => {
    const props = Object.assign(
      {
        title: undefined,
        description: undefined,
        titleColor: undefined,
        descriptionColor: undefined
      },
      propOverrides
    );
    const wrapper = mount(<MoheComponent {...props} />);
    return {
      props: props,
      component: wrapper.find("div"),
      anchor: wrapper.find("a")
    };
  };

  it("shoud always render div", () => {
    const { component } = setup();
    expect(component.length).toBeGreaterThan(0);
  });

  describe("Snapshots:", () => {
    it("should renders correctly", () => {
      const { component } = setup({
        title: "Title",
        description: "description"
      });
      expect(component).toMatchSnapshot();
    });
    it("should update the letters styles", () => {
      jest.useFakeTimers();
      const { component, anchor } = setup({
        title: "Title",
        description: "description",
        descriptionColor: "purple"
      });
      anchor.simulate("mouseEnter");
      jest.runAllTimers();
      expect(component).toMatchSnapshot();
    });

    it("should restore letters styles", () => {
      jest.useFakeTimers();
      const { component, anchor } = setup({
        title: "Title",
        description: "description"
      });
      anchor.simulate("mouseEnter");
      jest.runAllTimers();
      anchor.simulate("mouseLeave");
      jest.runAllTimers();
      expect(component).toMatchSnapshot();
    });
  });
});
