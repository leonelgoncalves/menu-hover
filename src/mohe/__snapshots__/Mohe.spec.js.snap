// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoheComponent: Snapshots: should renders correctly 1`] = `
<div>
  <a
    href="#"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <TextComponent
      text="Title"
      textRef={
        Object {
          "current": <span
            class=""
            aria-label="Title"
        >
            <span
                class="char1"
                aria-hidden="true"
            >
                T
            </span>
            <span
                class="char2"
                aria-hidden="true"
            >
                i
            </span>
            <span
                class="char3"
                aria-hidden="true"
            >
                t
            </span>
            <span
                class="char4"
                aria-hidden="true"
            >
                l
            </span>
            <span
                class="char5"
                aria-hidden="true"
            >
                e
            </span>
        </span>,
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "color": "",
          }
        }
      >
        Title
      </span>
    </TextComponent>
    <TextComponent
      text="description"
      textRef={
        Object {
          "current": <span
            class=""
        >
            description
        </span>,
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "color": "",
          }
        }
      >
        description
      </span>
    </TextComponent>
  </a>
</div>
`;

exports[`MoheComponent: Snapshots: should restore letters styles 1`] = `
<div>
  <a
    href="#"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <TextComponent
      text="Title"
      textRef={
        Object {
          "current": <span
            class=""
            aria-label="Title"
        >
            <span
                class="char1"
                aria-hidden="true"
                style="opacity: 1;"
            >
                T
            </span>
            <span
                class="char2"
                aria-hidden="true"
                style="opacity: 1;"
            >
                i
            </span>
            <span
                class="char3"
                aria-hidden="true"
                style="opacity: 1;"
            >
                t
            </span>
            <span
                class="char4"
                aria-hidden="true"
                style="opacity: 1;"
            >
                l
            </span>
            <span
                class="char5"
                aria-hidden="true"
                style="opacity: 1;"
            >
                e
            </span>
        </span>,
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "color": "",
          }
        }
      >
        Title
      </span>
    </TextComponent>
    <TextComponent
      text="description"
      textRef={
        Object {
          "current": <span
            class=""
        >
            description
        </span>,
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "color": "",
          }
        }
      >
        description
      </span>
    </TextComponent>
  </a>
</div>
`;

exports[`MoheComponent: Snapshots: should update the letters styles 1`] = `
<div>
  <a
    href="#"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <TextComponent
      text="Title"
      textRef={
        Object {
          "current": <span
            class=""
            aria-label="Title"
        >
            <span
                class="char1"
                aria-hidden="true"
                style="opacity: 0.3;"
            >
                T
            </span>
            <span
                class="char2"
                aria-hidden="true"
                style="opacity: 0.3;"
            >
                i
            </span>
            <span
                class="char3"
                aria-hidden="true"
                style="opacity: 0.3;"
            >
                t
            </span>
            <span
                class="char4"
                aria-hidden="true"
                style="opacity: 0.3;"
            >
                l
            </span>
            <span
                class="char5"
                aria-hidden="true"
                style="opacity: 0.3;"
            >
                e
            </span>
        </span>,
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "color": "",
          }
        }
      >
        Title
      </span>
    </TextComponent>
    <TextComponent
      text="description"
      textRef={
        Object {
          "current": <span
            class=""
            style="color: purple;"
        >
            description
        </span>,
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "color": "",
          }
        }
      >
        description
      </span>
    </TextComponent>
  </a>
</div>
`;
